Source/JavaScriptCore/assembler/MacroAssemblerX86_64.h

```diff
@@ -1247,6 +1247,19 @@ class MacroAssemblerX86_64 : public MacroAssemblerX86Common {
         m_assembler.xchgq_rm(src, dest.offset, dest.base);
     }
 
+    void swapDouble(FPRegisterID reg1, FPRegisterID reg2)
+    {
+        if (reg1 == reg2)
+            return;
+
+        // FIXME: This is kinda a hack since we don't use xmm7 as a temp.
+        ASSERT(reg1 != FPRegisterID::xmm7);
+        ASSERT(reg2 != FPRegisterID::xmm7);
+        moveDouble(reg1, FPRegisterID::xmm7);
+        moveDouble(reg2, reg1);
+        moveDouble(FPRegisterID::xmm7, reg2);
+    }
+
```

Comment:
Why pull this out into MacroAssemblerX86_64?  It looks identical to the x86 one.  Why not just have it as a common method in MacroAssemblerX86Common?

Reply:
Since we would like to remove MacroAssemblerX86Common.h eventually (since we no longer support 32bit x86 JIT), and we would like to move any methods we can move from that to X86_64 specific file.